cpu/kinetis_common/lpm_arch.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2014 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_kinetis_common_lpm
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the kernel's power management interface
+ *
+ * @author      Joakim Gebart <joakim.gebart@eistec.se>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "arch/lpm_arch.h"
+#include "kinetis_lpm.h"
+
+/*
+ * Counting semaphores for inhibiting undesired power modes.
+ */
+atomic_int_t kinetis_lpm_inhibit_stop_sema = {0};
+atomic_int_t kinetis_lpm_inhibit_vlps_sema = {0};
+atomic_int_t kinetis_lpm_inhibit_lls_sema  = {0};
+
+static inline void wait(void)
+{
+    /* Clear the SLEEPDEEP bit to make sure we go into WAIT (sleep) mode instead
+     * of deep sleep.
+     */
+    SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
+
+    /* Data synchronization barrier, wait for any outstanding writes to
+     * complete before entering low power mode. */
+    __DSB();
+    /* WFI instruction will start entry into WAIT mode */
+    __WFI();
+}
+
+/**
+ * @brief Switch the CPU into a STOP mode
+ *
+ * @param[in] stopmode     The chosen STOP mode
+ *
+ * @see Kinetis CPU reference manual, chapter System Mode Controller, Power Mode
+ * Control register (SMC_PMCTRL), field STOPM
+ */
+static inline void stop(uint8_t stopmode)
+{
+    volatile uint8_t dummy;
+    /* Set the SLEEPDEEP bit to enable deep sleep modes (STOP) */
```

Comment:
the function is over-commented, /* wait for outstanding writes and go into STOP (deep sleep) mode */ is enough.